pytorch_lightning/loops/dataloader/evaluation_loop.py

```diff
@@ -31,11 +32,15 @@ class EvaluationLoop(DataLoaderLoop):
 
     def __init__(self):
         super().__init__()
-        self._max_batches: Optional[Union[int, Sequence[int]]] = None
         self.outputs = []
+
         self.epoch_loop = EvaluationEpochLoop()
+
+        self.results = ResultCollection(training=False)
```

Comment:
are results meant to be mutable for external callers beyond the logger connector?

Reply:
Nope, removed the change